ls1intum/Artemis
# Enhancement/add option to programming submission service to load all pending submission for an exercise

### Checklist

### Motivation and Context



Improve the trigger button introduced in https://github.com/ls1intum/Artemis/pull/712.

### Description


This PR adds an option to load all pending submissions for a given programming exercise.
This is done to reduce the amount of REST calls when the participations page is entered (before 1 call per participation, now 1 call for the whole exercise).

A pending submission does not have a result yet and indicated after a certain time that the communication between Artemis, Bamboo, Bitbucket failed.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Open the participation area of a programming exercise **that has multiple participations** (e.g. #/exercise/747/participation)
4. Check the network tab - there should only be one call that retrieves the latest-pending-submission for each participation.

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/63600849-a8eee080-c5c4-11e9-938c-29155f894d91.png)
